community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/checkpoint/CountCommittedTransactionThreshold.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.transaction.log.checkpoint;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.neo4j.kernel.impl.util.Listener;
+
+public class CountCommittedTransactionThreshold implements CheckPointThreshold, TransactionCommitWatcher
```

Comment:
Wouldn't it be cheaper to not count transactions, but instead have `checkPointHappened()` accept a `transactionId`, keep that in a field here and just compare with that value in `newTransactionCommitted()`? I'm thinking that this may be an overhead since we scale well with concurrent committers and it feels unnecessary to introduce contention here.